### PR TITLE
Homework4

### PR DESCRIPTION
**Pull request by AlbertGaz**

### CODE REVIEW

File: hw/hw4/hw4_task3.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    """Write to stdout or stderr if starts with "error".
+
+    Args:
+        text: text to be printed
+
+    Returns: write to stderr if starts with error, else to stdout
+
+    """
+    if text.startswith("error"):
+        sys.stderr.write(text)
```
Comment (Aleksei-Samarin) [inline hw/hw4/hw4_task3.py]: Can we simplify these two blocks? They differs only by one element

File: hw/hw4/hw4_task4.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    >>> fizzbuzz(1.1)
+    Traceback (most recent call last):
+    ValueError: Input Error!
+    """
+    if type(n) != int or n < 1:
+        raise ValueError("Input Error!")
+    return [
+        "fizzbuzz"
+        if i % 15 == 0
+        else "fizz"
```
Comment (Aleksei-Samarin) [inline hw/hw4/hw4_task4.py]: These comprehensions are too complex here and below (namenum task_5), can we split them?

**Review by Aleksei-Samarin — CHANGES_REQUESTED** [review]

File: hw/hw4/hw4_task3.py (hunk 1 above)
Comment (AlbertGaz) [reply hw/hw4/hw4_task3.py]: Did you mean something like this?

dataflow = sys.stderr if text.startswith("error") else sys.stdout
dataflow.write(text)

File: hw/hw4/hw4_task4.py (hunk 2 above)
Comment (AlbertGaz) [reply hw/hw4/hw4_task4.py]: Done. Used idea that fizzbuzz is string composed from fizz + buzz

**Pull request merged**